heatmap/src/heatmap.rs

```diff
@@ -201,20 +206,37 @@ impl Heatmap {
     /// values.
     fn tick(&self, time: Instant) {
         loop {
```

Comment:
This is an infinite loop if the lock gets poisoned somehow.

Is there a reason you're doing a busy loop here instead of something like the code below?
```rust
if time < self.next_tick.load(Ordering::Relaxed)  {
  return;
}

let _lock = self.lock.lock().expect("Lock was poisoned"); // or unpoison the lock here
if time < self.next_tick.load(Ordering::Relaxed) {
  return;
}

...
```

Reply:
I've replaced the mutex with the one from `parking_lot` to avoid the issue with the lock being poisoned. Added another check after acquiring the lock to make sure we still need to tick forward after lock acquisition. Expanded the comments a bit. It is expected that in the common case where we've failed to lock, that another thread has already ticked the heatmap forward enough for the loop to return when we check next_tick again.